src/tree_schema_compile.c

```diff
@@ -2770,10 +2770,10 @@ lys_compile_type_(struct lysc_ctx *ctx, struct lysp_node *context_node_p, uint16
             lref->require_instance = 1;
         }
         if (type_p->path) {
-            lref->path = lyxp_expr_dup(ctx->ctx, type_p->path);
+            lyxp_expr_dup(ctx->ctx, type_p->path, &lref->path);
```

Comment:
isn't the point of the change to be able to check for the error return code of the modified functions? The same applies for all the changes in this file.

Reply:
Yes, for cases where the error was checked, I did not change the functionality. But okay, it is checked now and the whole function was refactored because the error handling was weird (probably outdated).